docs/howto/object-storage/s3/object-lock.md

```diff
@@ -0,0 +1,284 @@
+# Object lock
+
+Object Lock is a feature in S3 that enables you to lock your objects
+to prevent them from being deleted or overwritten. This feature provides
+a storage solution based on the **Write Once, Read Many (WORM)** model,
+which can be helpful when you want to maintain data integrity,
+comply with regulations, or retain data for a specific period.
+
+> Are you looking to set the expiration/retention time for your
+> objects? Then you should check our [Object expiry](expiry.md) guide.
+
+## Object Lock modes
+
+Object Lock offers two methods for managing object retention:
+**Retention periods** and **Legal hold**.
+
+### Retention periods
+
+Retention periods specify the length of time that an object
+remains locked and protected from being overwritten or deleted.
+When you set a retention period, the object is safeguarded for
+the specified time period. You can set the retention period in
+either days or years, with a **minimum of one day** and
+no maximum limit.
+
+In addition to retention periods, you can choose the retention
+mode that applies to your objects. At the moment only the
+**Compliance** mode is available.
```

Comment:
I prefer "currently" over "at the moment", because the former is neutral and the latter hints that something is about to change soon.

Also, if of the two object lock retention modes we support only one, we need an entry to that effect in the object storage of the Limitations reference (`docs/reference/limitations/object-storage.md`).

Reply:
Saw that you already took care of this, thanks :)